### DISCUSSION TITLE
KafkaExporter to monitor consumer lags

### DISCUSSION THREAD

**Discussion by pashafirdous** [strimzi/strimzi-kafka-operator#4141] [General]
Hi ,

Could you please advice on steps to access metrics from Kafka exporter on port 9404?

when deploye in 0.17.0 we got a service for kafka exporter to access metrics on http://kafkaexporter-service:9404/metrics.
we are tryingthe same with 0.20.0/0.20.1.

we understand the monitoring is changes in these version, ut please advice how can  acccss the monitoring endpoint.

**Comment by scholzj** [comment]
You just have to scrape the metrics from the pod directly and not through the service. How exactly to do it depends on how you deployed and manage your Prometheus. For example if you use the Prometheus Operator, you can use the PodMonitor resources for it: https://github.com/strimzi/strimzi-kafka-operator/blob/0.20.1/examples/metrics/prometheus-install/strimzi-pod-monitor.yaml

**Reply by pashafirdous** [reply]
we dont want to deploy Prometheus as we use a 3rd party application to scrape the prometheus metrics. the earlier version i was able to do it via service. 

can we enable the Kafka-exporter he way we see Kafka/zookeeper?

**Reply by scholzj** [reply]
The Kafka Exporter works exactly the same way as Kafka or Zookeeper. The 9404 port is exposed only on the pods. I think you have two options ... first of all, you can ask your 3rd party application provider how to do it. It is not unique feature of Strimzi, so I would assume they should be able to do it. As an alternative, you can manually create services with 9404 port exposed - just name them differently then the services created by the operator but point them to the same pods. That way you can manully restore similar state as was in previous versions.

**Reply by pashafirdous** [reply]
perfect. it worked.
